server/lib/nl/counters.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Dict
+
+
+#
+# A helper class to track info / error / timing counters.
+#
+class Counters:
+
+  def __init__(self):
+    self._warn: Dict = {}
+    self._info: Dict = {}
+    self._timing: Dict = {}
+
+  #
+  # Adds/updates an info counter.
+  #
+  # For a given counter, caller should always pass the same type
+  # for value.  If value is numeric, then its a single updated
+  # counter, otherwise, counter is a list of values.
+  #
+  def info(self, counter: str, value: any):
```

Comment:
This is updating the counter so why not have the function name reflect that? info() seems to imply you're getting some information instead of updating an info counter, IMHO. I might even suggest `update_info()` or `add_info()`? If you agree, then the same would apply to `warn()` and `timeit()` below.

Reply:
As you see this gets called from all over the fulfillment code, so I was trying to use as few chars as possible :)

In most places, we already have `state.uttr.counters.info(...)` which is already a mounthful?

Tried to model like `logging.info()`, rather than `logging.add_info()`.